src/modals/ForceUndelegateModal.tsx

```diff
@@ -0,0 +1,263 @@
+import React, { useMemo, useState } from 'react'
+import moment from 'moment'
+import styled from 'styled-components'
+import JiraFailedBuildStatusIcon from '@atlaskit/icon/glyph/jira/failed-build-status'
+import { truncateStreamName } from '~/shared/utils/text'
+import FormModal, {
+    FormModalProps,
+    FormModalRoot,
+    SectionHeadline,
+} from '~/modals/FormModal'
+import { Tip, TipIconWrap } from '~/components/Tip'
+import { BN } from '~/utils/bn'
+import { fromAtto } from '~/marketplace/utils/math'
+import { abbreviateNumber } from '~/shared/utils/abbreviateNumber'
+import { ScrollTable } from '~/shared/components/ScrollTable/ScrollTable'
+import { Alert } from '~/components/Alert'
+import { Radio } from '~/shared/components/Radio'
+
+interface Props extends Omit<FormModalProps, 'canSubmit' | 'onSubmit'> {
+    onResolve?: (sponsorshipId: string) => void
+    onSubmit: (sponsorshipId: string) => Promise<void>
+    tokenSymbol: string
+    sponsorships: UndelegateSponsorship[]
+    totalAmount: BN
+}
+
+interface UndelegateSponsorship {
+    id: string
+    streamId: string
+    amount: BN
+    minimumStakingPeriodSeconds: number
+    joinTimestamp: number
+}
+
+function getOptimalSponsorship(
+    sponsorships: UndelegateSponsorship[],
+    requestedAmount: BN,
+): UndelegateSponsorship | undefined {
+    return sponsorships.find(
+        (s) =>
+            hasStakedLongEnough(s.joinTimestamp, s.minimumStakingPeriodSeconds) &&
+            s.amount.isGreaterThanOrEqualTo(requestedAmount),
+    )
+}
+
+function hasStakedLongEnough(joinTimestamp: number, minimumStakingPeriodSeconds: number) {
+    const joinDate = moment(joinTimestamp * 1000)
+    const slashedBeforeDate = joinDate.clone().add(minimumStakingPeriodSeconds, 'seconds')
+    return moment().isAfter(slashedBeforeDate)
+}
+
+export default function ForceUndelegateModal({
+    title = 'Force unstake',
+    onResolve,
+    onSubmit,
+    submitLabel = 'Force unstake',
+    tokenSymbol,
+    sponsorships: unsortedSponsorships,
+    totalAmount,
+    ...props
+}: Props) {
+    const sponsorships = useMemo(
+        () => [...unsortedSponsorships].sort((a, b) => b.amount.comparedTo(a.amount)),
+        [unsortedSponsorships],
+    )
+
+    const [selectedSponsorshipId, setSelectedSponsorshipId] = useState<
+        string | undefined
+    >(
+        getOptimalSponsorship(sponsorships, totalAmount)?.id ??
```

Comment:
If `id` is and empty string, `''` will be the new `selectedSponsorshipId` (`??` magic kicked in).

```ts
getOptimalSponsorship(sponsorships, totalAmount)?.id || sponsorships[0]?.id
```

should be enough.